pkg/cloud/services/eks/nodegroup.go

```diff
@@ -591,12 +590,7 @@ func (s *NodegroupService) setStatus(ng *eks.Nodegroup) error {
 		for _, group := range groups.AutoScalingGroups {
 			replicas += int32(len(group.Instances))
 			for _, instance := range group.Instances {
-				id, err := noderefutil.NewProviderID(fmt.Sprintf("aws://%s/%s", *instance.AvailabilityZone, *instance.InstanceId))
-				if err != nil {
-					s.Error(err, "couldn't create provider ID for instance", "id", *instance.InstanceId)
-					continue
-				}
-				providerIDList = append(providerIDList, id.String())
+				providerIDList = append(providerIDList, fmt.Sprintf("aws:///%s/%s", *instance.AvailabilityZone, *instance.InstanceId))
```

Comment:
Hmm, I would prefer using safe dereferences here. If, for whatever reason, AZ or InstanceID is nil, this will panic.

Reply:
In any case, these two fields should not be empty as these are required parameters.
Do you think adding validations would help here? I don't strongly feel the need for validation, but I am ok either ways 😄

Reply:
Oh, I just thought something like using `aws.String` instead of dereferencing. Nothing too fancy. :)

Reply:
Just to prevent a possible panic. An error is much preferred I believe? :) Because a panic stops events from happening such as deletion for example.

Reply:
Oh so sorry i misunderstood it as validity check for dereferencing. Yes you are right, we could use aws library functions for dereferencing.
cc @sedefsavas could you please take care of this comment?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3747#discussion_r979633728

Can we target this after merging this PR? We might change the providerID string as per Richard's suggestion.

Reply:
Sounds good, also whatever change we do here, we need to do on the unmanaged side as well.